code/modules/mechs/components/_components.dm

```diff
@@ -101,27 +109,91 @@
 	to_chat(user, SPAN_NOTICE(" - Integrity: <b>[round((((max_damage - total_damage) / max_damage)) * 100)]%</b>" ))
 
 /obj/item/mech_component/attackby(obj/item/I, mob/living/user)
-	if(I.use_tool(user, src, WORKTIME_INSTANT, QUALITY_SCREW_DRIVING, FAILCHANCE_ZERO))
-		if(contents.len)
-			//Filter non movables
-			var/list/valid_contents = list()
-			for(var/atom/movable/A in contents)
-				if(!A.anchored)
-					valid_contents += A
-			if(!valid_contents.len)
-				return
-			var/obj/item/removed = pick(valid_contents)
-			if(!(removed in contents))
-				return
-			if(eject_item(removed, user))
-				update_components()
-		else
-			to_chat(user, SPAN_WARNING("There is nothing to remove."))
+	var/list/usable_qualities = list(QUALITY_PULSING, QUALITY_BOLT_TURNING, QUALITY_WELDING, QUALITY_PRYING, QUALITY_SCREW_DRIVING)
+	var/tool_type = I.get_tool_type(user, usable_qualities, src)
+
+	switch(tool_type)
+		if(QUALITY_SCREW_DRIVING)
+			if(contents.len)
+				//Filter non movables
+				var/list/valid_contents = list()
+				for(var/atom/movable/A in contents)
+					if(!A.anchored)
+						valid_contents += A
+				if(!valid_contents.len)
+					return
+				var/obj/item/removed = pick(valid_contents)
+				if(!(removed in contents))
+					return
+				if(eject_item(removed, user))
+					update_components()
+			else
+				to_chat(user, SPAN_WARNING("There is nothing to remove."))
+
+		if(QUALITY_WELDING)
```

Comment:
```suggestion
		if(QUALITY_WELDING && user.a_intent == I_HELP)
```
Otherwise you will repair the mech even if you want to attack it with welder in harm mode

Reply:
uh?? maybe I'm tired but this appears to be for when the component is laying on the floor not for the mech itself

Reply:
Ah, alright. I thought it was triggered when you clicked on the mech (like mech receives its attackby and transfer it to the component the user is aiming it)